views.py

```diff
@@ -1,47 +1,86 @@
 from flask import Blueprint, render_template, request, redirect, url_for
+import json
+import os
+
 
 main = Blueprint('main', __name__)
 
-items = []
+
 @main.route('/')
 def index():
-	return render_template('homepage.html', items=items)
+	#with open('list.json') as file:
+	#	items = json.load(file)
+	with open('lists.json') as file:
+		try:
+			lists = json.load(file)
+		except:
+			lists = ''
+	return render_template('homepage.html', lists=lists)
 
-@main.route('/list')
+@main.route('/list', methods=['POST'])
 def list():
-	return render_template('list.html', items=items)
-
-@main.route('/remove', methods=['POST'])
-def list_remove():
-	list_remove = request.form.get('removes')
-	if list_remove:
-		items.remove(f'{list_remove}')
-	return render_template('homepage.html', items=items)
+	name = request.form.get('chooselist')
+	with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
+		items = json.load(file)
+	return render_template('list.html', items=items, name=name)
 
 @main.route('/list_post', methods=['POST'])
 def list_post():
+	name = request.form.get('names')
+	with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
+		items = json.load(file)
 	list_item = request.form.get('items')
 	if list_item:
 		items.append(f'{list_item}')
-	return render_template('homepage.html',items=items)
+	with open('/Users/austin/src/web_app/lists/' + name + '.json', 'w') as file:
+		json.dump(items, file)
```

Comment:
Same as above. You could have a generic function to save your updated items.

```
def saveItems(name, items):
    with open('/Users/austin/src/web_app/lists/' + name + '.json') as file:
		json.dump(items, file)
```


```
saveItems(name, items)
```

And same as above, anywhere you're saving items you can just use this function.